lib/nahmii-provider.js

```diff
@@ -47,23 +49,68 @@ class NahmiiProvider extends ethers.providers.JsonRpcProvider {
         throw new Error('Not implemented');
     }
 
+    /**
+     * Force provider to start updating the API access token once a minute. The
+     * update will continue until stopUpdate() is called.
+     * Using any methods that require an API access token will automatically
+     * start the update process.
+     */
+    startUpdate() {
+        if (this.isUpdating)
+            return;
+
+        const updateToken = () => {
+            console.log('Updating API access token...');
+            const baseUrl = _baseUrl.get(this);
+            const appId = _appId.get(this);
+            const appSecret = _appSecret.get(this);
+
+            _apiAccessTokenPending.set(this, createApiToken(baseUrl, appId, appSecret)
+                .then(t => {
+                    console.log('Successfully updated API access token.');
+                    _apiAccessToken.set(this, t);
+                    return t;
+                })
+                .catch(err => {
+                    console.log('Failed to update API access token!');
+                    return Promise.reject(err);
+                }));
+        };
+
+        updateToken();
+        _tokenInterval.set(this, setInterval(updateToken, 60 * 1000));
+    }
+
+    /**
+     * Stops the automatic update of API access tokens. This should be called
+     * when you want to terminate/delete the provider to ensure there are no
+     * lingering references.
+     */
+    stopUpdate() {
+        clearInterval(_tokenInterval.get(this));
+        _tokenInterval.set(this);
+    }
+
+    /**
+     * Returns the state of the API access token update process.
+     * @returns {boolean} True if the update is running, false otherwise.
+     */
+    get isUpdating() {
+        return !!_tokenInterval.get(this);
+    }
+
     /**
      * Resolves into the current token, or will obtain a new token from the
      * server as needed.
      * @returns {Promise} A promise that resolves into an API access token
      */
     async getApiAccessToken() {
```

Comment:
Do you still need the `async` keyword?

Reply:
Yes because both the pending token and the current token should be returned as promises.